### PR TITLE
[WIP] Add jsdoc to project

### LINKED ISSUE #1739: Wiki and jsDoc
Description:

It's gonna take some time but I really think we should start building up our wiki / dump some doc from other sources there. 
Also just a list of references of things you should know may be enough to start (also linking all the cosmos and tendermint docs etc.)

Supports #1694

### PR DESCRIPTION
**Pull request by sabau**
_Description:_
- update minor version of vue
- examples of documentation in some modules
- added command `doc` to package.json
- explicitely working only on app/src files

Closes #1739

Once we are in a presentable state we may link this in the wiki along with our README

❤️ Thank you!

---

- [ ] Added entries in `CHANGELOG.md` with issue # and GitHub username
- [ ] Reviewed `Files changed` in the github PR explorer

### CODE REVIEW

File: docs/cosmos-voyager/0.8001.1/module-App.html [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
+</head>
+
+<body>
+
+<div id="main">
+
+    <h1 class="page-title">Module: App</h1>
+
+
```
Comment (fedekunze) [inline docs/cosmos-voyager/0.8001.1/module-App.html]: why so many empty lines on this file ?

Comment (sabau) [reply docs/cosmos-voyager/0.8001.1/module-App.html]: It's autogenerated, probably jsdoc leave some space for extra informations in a static manner

File: app/src/renderer/routes.js [hunk 2]
```diff
@@ -59,32 +58,32 @@ export default [
...
   {
     path: `/transactions`,
     name: `transactions`,
-    component: wallet(`Transactions`)
+    component: require(`./components/wallet/PageTransactions`).default
   },
 
-  { path: `/404`, component: common(`404`) },
-  { path: `*`, component: common(`404`) }
+  { path: `/404`, component: require(`./components/common/Page404`).default },
```
Comment (faboweb) [inline app/src/renderer/routes.js]: nice

**Comment by sabau** [comment]
Usually is more convenient to have the doc in the project, to have it more accessible, it can be also be compressed for older versions.

We could enforce it on cci of via git hook, as for now the command to update it manually:

```
yarn doc
```

**Comment by faboweb** [comment]
OK cool, going with it for now.

**Comment by fedekunze** [comment]
do we need to update the `README` ?

**Comment by faboweb** [comment]
> do we need to update the README ?

good point

**Comment by faboweb** [comment]
Let's halt the merge until further discussion

**Comment by sabau** [comment]
I closed this one to have a cleaner single commit, just squashed the changes into another branch

**Pull request closed without merging**